From 39d6f37aef66fcd1a1caaf5ea2bcf8a3a3ea5684 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sat, 08 Feb 2020 21:08:01 +0200
Subject: [PATCH] [SW-8229] goya2: fix 64bit MSB mask

Mask was for lower 16 bit, instead of masking all lower 32bit.

Change-Id: I507a0ee0c36b18d754094c29dd050920d8344e95
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/dma.h b/common/dma.h
index 9907966..ef92a5f 100644
--- a/common/dma.h
+++ b/common/dma.h
@@ -17,7 +17,7 @@
 #define DMA_DST_BLOCK_SIZE	4
 
 #define DMA_ADDRESS_LOW(x)	((x) & 0xFFFFFFFF)
-#define DMA_ADDRESS_HIGH(x)	(((x) >> 32) & 0xFFFF)
+#define DMA_ADDRESS_HIGH(x)	(((x) >> 32) & 0xFFFFFFFF)
 
 void dma_copy_image(uint16_t image_size, uint32_t flash_offset,
 		    uint64_t dest_offset, uint8_t fw_sram_dst);
